### ISSUE TITLE
need a way to resize a mid-list item without dropping scroll position

### ISSUE THREAD

**Issue by danbeam** [PolymerElements/iron-list#116]
I'm trying to incorporate `<iron-list>` into the new Chrome downloads UI.

The current implementation looks something like this:

```html
  <iron-list id="downloads-list" items="{{items_}}">
    <template>
      <downloads-item data="[[item.item]]">
    </template>
  </iron-list>
```

and the JS populates the list [kind of] like this:

```js
/** Called on page load. */
updateAllItems: function(items) {
  this.items_ = items.map(function(item, index) {
    return {item: item, index: index};
  });
},

/**
 * Called multiple times a second when an item is downloading
 * i.e. to show "500KB of 5MB, 250KB/s", and update a <paper-progress>.
 */
updateSingleItem: function(item) {
  this.set('items_.' + this.getIndex_(item) + '.item', item);
},
```

`updateSingleItem` can cause the items to change in height.  When an item finishes, it shrinks by 20-30px in height, and we end up with this:

![in progress](http://i.imgur.com/bXzHy3x.png)
![finished](http://i.imgur.com/19ti4cA.png)

To fix this, I added:

```js
updateSingleItem: function(item) {
  this.set('items_.' + this.getIndex_(item) + '.item', item);
+ this.$['downloads-list'].fire('iron-resize');
},
```

But this resets scroll position.  While a download is progressing, an item is updated every ~500ms, which constantly scrolls to the top and makes the page unusable.

**tl;dr - how can I update an item without reseting the scroll position or leaving gaps in the list?**

/cc @tjsavage @blasten

**Comment by blasten** [comment]
@danbeam Agreed. I have a branch https://github.com/PolymerElements/iron-list/pull/45 that attempts to solve this problem and it also supports animations on items height. I can work on a new PR based on that old PR only with the method `updateSizeForIndex(idx)`

**Comment by blasten** [comment]
merged now!
